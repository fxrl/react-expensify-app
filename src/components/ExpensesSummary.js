import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { getExpensesTotal } from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

const ExpensesSummary = (props) => {
    const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
    const total = getExpensesTotal(props.expenses);
    const formattedTotal = numeral(total / 100).format('$0,0.0000');
    return (
        <div>
            <h1> Viewing {props.expenses.length} {expenseWord} totalling {formattedTotal} </h1>
        </div>
    );
}

const mapStateToProps = (state) => {
    return {expenses: selectExpenses(state.expenses, state.filters)};
};

export default connect(mapStateToProps)(ExpensesSummary);