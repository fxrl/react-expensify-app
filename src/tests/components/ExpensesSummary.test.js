import { shallow } from 'enzyme';
import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses'

test('should render expenses summary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render one expense on expense summary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]}/>);
    expect(wrapper).toMatchSnapshot();
})