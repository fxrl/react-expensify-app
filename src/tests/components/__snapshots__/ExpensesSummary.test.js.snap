// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary correctly 1`] = `
<div>
  <h1>
     Viewing 
    3
     
    expenses
     totalling 
    $1,135.9500
     
  </h1>
</div>
`;

exports[`should render one expense on expense summary correctly 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $678.0000
     
  </h1>
</div>
`;
